mitmproxy/addons/proxyserver.py

```diff
@@ -121,15 +129,48 @@ async def shutdown_server(self):
         self.server = None
 
     async def handle_connection(self, r, w):
+        peername = w.get_extra_info('peername')
         asyncio_utils.set_task_debug_info(
             asyncio.current_task(),
             name=f"Proxyserver.handle_connection",
-            client=w.get_extra_info('peername'),
+            client=peername,
         )
         handler = ProxyConnectionHandler(
             self.master,
             r,
             w,
             self.options
         )
-        await handler.handle_client()
+        self._connections[peername] = handler
+        try:
+            await handler.handle_client()
+        finally:
+            del self._connections[peername]
+
+    def inject_event(self, flow: flow.Flow, event: events.Event):
+        if flow.client_conn.peername not in self._connections:
+            raise ValueError("Flow is not from a live connection.")
+        self._connections[flow.client_conn.peername].server_event(event)
+
+    @command.command("inject")
+    def inject(self, flows: Sequence[flow.Flow], from_client: bool, message: str):
```

Comment:
What is the intention of having `Sequence[flow.Flow]`? Is there a remotely realistic scenario where someone injects the exact same thing into multiple flows? That would require keeping references for more than one flow, when usually you process them individually in add-ons. I think this makes the call to `inject` awkward.